databricks/koalas/tests/test_dataframe.py

```diff
@@ -4546,6 +4546,17 @@ def test_explain_hint(self):
         finally:
             sys.stdout = prev
 
+    def test_repartition(self):
+        kdf = ks.DataFrame({"age": [5, 5, 2, 2], "name": ["Bob", "Bob", "Alice", "Alice"]})
+
+        new_kdf = kdf.spark.repartition(7)
+        self.assertEqual(new_kdf.to_spark().rdd.getNumPartitions(), 7)
+
+        kdf = kdf.set_index("age")
+        new_kdf = kdf.spark.repartition(5)
+        self.assertEqual(new_kdf.to_spark().rdd.getNumPartitions(), 5)
+        self.assertEqual(kdf.index.name, new_kdf.index.name)
```

Comment:
Shall we put this test in `test_frame_spark`?

Reply:
Sounds good!